core/src/main/java/org/apache/calcite/sql/dialect/SqliteSqlDialect.java

```diff
@@ -40,9 +41,30 @@ public class SqliteSqlDialect extends SqlDialect {
 
   public static final SqlDialect DEFAULT = new SqliteSqlDialect(DEFAULT_CONTEXT);
 
+  private final int majorVersion;
+  private final int minorVersion;
+
   /** Creates a SqliteSqlDialect. */
   public SqliteSqlDialect(SqlDialect.Context context) {
     super(context);
+    this.majorVersion = context.databaseMajorVersion();
+    this.minorVersion = context.databaseMinorVersion();
+  }
+
+  @Override public boolean supportsJoinType(JoinRelType joinType) {
+    // Unknown version means we conservatively assume support for all join types
```

Comment:
Ww should do the safest thing. Is that allowing queries with join types that the engine can't execute or blocking legal queries when we don't know the version? I am not sure what's best here tbh.